src/modules/ce/components/AssignContactsButton.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  Button,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Stack,
+} from '@mui/material';
+import { reduce } from 'lodash-es';
+import React, { useCallback, useMemo, useState } from 'react';
+import AssignContactFormItem from './AssignContactFormItem';
+import ButtonTooltipContainer from '@/components/elements/ButtonTooltipContainer';
+import CommonDialog from '@/components/elements/CommonDialog';
+import { ContactsIcon } from '@/components/elements/SemanticIcons';
+import FormDialogActionContent from '@/modules/form/components/FormDialogActionContent';
+import {
+  CeReferralFieldsFragment,
+  useAssignParticipantsMutation,
+} from '@/types/gqlTypes';
+
+interface Props {
+  referral: CeReferralFieldsFragment;
+}
+const AssignContactsButton: React.FC<Props> = ({ referral }: Props) => {
+  const [open, setOpen] = useState(false);
+
+  const swimlanes = referral.opportunity.swimlanes || [];
+
+  // Map the existing referral participants to a Record<string, string[]>
+  // where key = swimlaneId and value = [userIds]
+  const initialValues = useMemo(() => {
+    return reduce(
+      referral.participants,
+      (acc, { swimlane, user }) => ({
+        ...acc,
+        [swimlane.id]: [...(acc[swimlane.id] || []), user.id],
+      }),
+      {} as Record<string, string[]>
+    );
+  }, [referral.participants]);
+
+  // `values` maps swimlaneId to [userIds]
+  const [values, setValues] = useState<Record<string, string[]>>(initialValues);
+
+  const handleChangeValue = useCallback(
+    (swimlaneId: string, userIds: string[]) => {
+      setValues((prevValues: Record<string, string[]>) => ({
+        ...prevValues,
+        [swimlaneId]: userIds,
+      }));
+    },
+    [setValues]
+  );
+
+  const [handleSubmit, { loading, error }] = useAssignParticipantsMutation({
+    variables: {
+      referralId: referral.id,
+      // On submit, map the `values` object into the shape the mutation is expecting: { userId, swimlaneId }
+      participants: Object.entries(values).flatMap(([swimlaneId, userIds]) =>
+        userIds.map((userId) => ({ userId, swimlaneId }))
+      ),
+    },
+    onCompleted: () => setOpen(false),
+  });
+
+  if (error) throw error;
+
+  return (
+    <>
+      {swimlanes.length === 0 ? (
+        // If this referral has no swimlanes, disable the button.
+        // For now this will only happen if the referral also has no tasks,
+        // because the schema requires swimlane to be non-null on tasks.
```

Comment:
Let's discuss this. To expand on the comment a bit: The _database_ does not require swimlanes to exist for a template. But the _schema_ does require swimlane to exist on a task. So if the workflow has any tasks, this code would never run, because if it had tasks but no swimlanes, you'd see a schema error "cannot return null for non-nullable field."

However, you can see this code in action if you create a workflow with the template "No Tasks" (succeeds immediately).

Reply:
This seems OK to me.  Now that the schema is updated, the check on line 65 could be simplified to `referral.swimlanes.length === 0`.